src/external/electrs.ts

```diff
@@ -250,14 +267,30 @@ export class DefaultBTCCoreAPI implements BTCCoreAPI {
                 continue;
             }
             for (const vout of tx.vout) {
-                if (this.txOutputHasRecipientAndAmount(vout, recipientAddress, amountAsBTC)) {
+                if (this.txOutputHasRecipientAndAmount(vout, recipientAddress, amount)) {
                     return tx.txid;
                 }
             }
         }
         return Promise.reject("No transaction id found");
     }
 
+    async waitForOpreturn(data: string, timeoutMs: number, retryIntervalMs: number): Promise<string> {
+        return new Promise<string>((resolve, reject) => {
+            this.getTxIdByOpReturn(data)
+                .then(resolve)
+                .catch((_error) => {
+                    setTimeout(() => {
+                        console.log("Did not find opreturn, retrying...");
+                        if(timeoutMs < retryIntervalMs) {
+                            reject("Timeout elapsed");
+                        }
+                        this.waitForOpreturn(data, timeoutMs - retryIntervalMs, retryIntervalMs).then(resolve);
+                    }, 5000);
```

Comment:
shouldn't the 5000 be the `retryInterval`

Reply:
maybe this should also be a constant?